File: frontend/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  description: string;
  price: number;
  status: 'ACTIVE' | 'INACTIVE';
  imageUrl?: string;
}

export interface ProductResponse {
  id: string;
  name: string;
  description: string;
  price: number;
  status: string;
  imageUrl?: string;
}

File: backend/src/main/java/com/shopcart/product/entity/Product.java
package com.shopcart.product.entity;

import com.shopcart.common.enums.ProductStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "products")
public class Product{
    @Id
    @Column(name = "product_id",length = 50)
    private String id;
    
    @Column(nullable= false)
    private String name;
    
    @Column(nullable = false)
    private Long price;
    
    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ProductStatus status;
}
